azure-iot-device/azure/iot/device/iothub/pipeline/pipeline_stages_iothub_mqtt.py

```diff
@@ -51,32 +51,33 @@ def _run_op(self, op):
 
             # Apply query parameters (i.e. key1=value1&key2=value2...&keyN=valueN format)
             custom_product_info = str(self.pipeline_root.pipeline_configuration.product_info)
-            if custom_product_info.startswith(
-                pkg_constant.DIGITAL_TWIN_PREFIX
-            ):  # Digital Twin Stuff
-                query_param_seq.append(("api-version", pkg_constant.DIGITAL_TWIN_API_VERSION))
+
+            if self.pipeline_root.pipeline_configuration.model_id is not None:
+                # Digital Twin Stuff
+                query_param_seq.append(
+                    ("api-version", pkg_constant.IOTHUB_DIGITAL_TWIN_API_VERSION)
+                )
                 query_param_seq.append(("DeviceClientType", user_agent.get_iothub_user_agent()))
                 query_param_seq.append(
-                    (pkg_constant.DIGITAL_TWIN_QUERY_HEADER, custom_product_info)
+                    ("model-id", self.pipeline_root.pipeline_configuration.model_id)
                 )
             else:
                 query_param_seq.append(("api-version", pkg_constant.IOTHUB_API_VERSION))
-                query_param_seq.append(
-                    ("DeviceClientType", user_agent.get_iothub_user_agent() + custom_product_info)
-                )
+                query_param_seq.append(("DeviceClientType", user_agent.get_iothub_user_agent()))
 
             # NOTE: Client ID (including the device and/or module ids that are in it)
             # is NOT url encoded as part of the username. Neither is the hostname.
             # The sequence of key/value property pairs (query_param_seq) however, MUST have all
             # keys and values URL encoded.
             # See the repo wiki article for details:
             # https://github.com/Azure/azure-iot-sdk-python/wiki/URL-Encoding-(MQTT)
-            username = "{hostname}/{client_id}/?{query_params}".format(
+            username = "{hostname}/{client_id}/?{query_params}{product_info}".format(
                 hostname=self.pipeline_root.pipeline_configuration.hostname,
                 client_id=client_id,
                 query_params=version_compat.urlencode(
                     query_param_seq, quote_via=urllib.parse.quote
                 ),
+                product_info=urllib.parse.quote(custom_product_info, safe=""),
```

Comment:
This is a clarity change - functionally it does the same thing it was always doing, but it's a lot more readable now, and results in less immediate questions, such as "wait, why was custom product info being appended to the end of the user agent query parameter on line 65????")